core/peerset/src/lib.rs

```diff
@@ -646,5 +651,40 @@ mod tests {
 
 		futures::executor::block_on(fut);
 	}
+
+
+	#[test]
+	fn test_peerset_reserved_node_not_banned() {
+		let bootnode = PeerId::random();
+		let reserved_peer = PeerId::random();
+		let config = PeersetConfig {
+			in_peers: 0,
+			out_peers: 1,
+			bootnodes: vec![bootnode],
+			reserved_only: true,
+			reserved_nodes: Vec::new(),
```

Comment:
this might need to be
```rust
reserved_nodes: vec![reserved_peer],
```

Reply:
Just refer to test case `test_peerset_add_reserved_peer`, not sure the difference between `handle.add_reserved_peer()` and `reserved_nodes: vec![reserved_peer]`